src/Microsoft.DotNet.Wpf/src/Shared/MS/Utility/FrugalList.cs

```diff
@@ -1298,111 +1292,69 @@ public ArrayItemList(int size)
-                _entries = new T[_count];
-                collection.CopyTo(_entries, 0);
-            }
+            Debug.Assert(collection is not null);
+            _count = collection.Count;
+            _entries = new T[_count];
+            collection.CopyTo(_entries, 0);
         }
 
         // Capacity of this store
-        public override int Capacity
-        {
-            get
-            {
-                if (_entries != null)
-                {
-                    return _entries.Length;
-                }
-                return 0;
-            }
-        }
+        public override int Capacity => _entries.Length;
 
         public override FrugalListStoreState Add(T value)
         {
             // If we don't have any entries or the existing entry is being overwritten,
             // then we can use this store. Otherwise we have to promote.
-            if ((null != _entries) && (_count < _entries.Length))
+            if (_count < _entries.Length)
             {
                 _entries[_count] = value;
                 ++_count;
             }
             else
             {
-                if (null != _entries)
-                {
-                    int size = _entries.Length;
+                int size = _entries.Length;
 
-                    // Grow the list slowly while it is small but
-                    // faster once it reaches the LARGEGROWTH size
-                    if (size < LARGEGROWTH)
-                    {
-                        size += GROWTH;
-                    }
-                    else
-                    {
-                        size += size >> 2;
-                    }
-
-                    T[] destEntries = new T[size];
-
-                    // Copy old array
-                    Array.Copy(_entries, 0, destEntries, 0, _entries.Length);
-                    _entries = destEntries;
+                // Grow the list slowly while it is small but
+                // faster once it reaches the LARGEGROWTH size
+                if (size < LARGEGROWTH)
+                {
+                    size += GROWTH;
                 }
                 else
                 {
-                    _entries = new T[MINSIZE];
+                    size += size >> 2;
                 }
 
+                T[] destEntries = new T[size];
+
+                // Copy old array
+                Array.Copy(_entries, 0, destEntries, 0, _entries.Length);
+                _entries = destEntries;
+
                 // Insert into new array
                 _entries[_count] = value;
                 ++_count;
             }
             return FrugalListStoreState.Success;
         }
 
-        public override void Clear()
-        {
-            // Wipe out the info.
-            for (int i = 0; i < _count; ++i)
-            {
-                _entries[i] = default(T);
-            }
-            _count = 0;
-        }
+        public override void Clear() => _entries.AsSpan(0, _count).Clear();
```

Comment:
```suggestion
        public override void Clear()
        {
            _entries.AsSpan(0, _count).Clear();
            _count = 0;
        }
```

I haven't tried applying this fix to my custom build, but the missing assignment to `_count` seems very likely to be the cause of the crash I reported.